sdk/src/main/java/com/google/cloud/dataflow/sdk/runners/DataflowPipelineRunner.java

```diff
@@ -420,6 +426,37 @@ public static DataflowPipelineRunner fromOptions(PipelineOptions options) {
     return super.apply(new AssignWindows<>(transform), input);
   }
 
+  @Nullable
+  private void maybeRegisterDebuggee(DataflowPipelineOptions options, String uniquifier) {
```

Comment:
The pattern of `void maybeDoSomething(flags_that_control_will_the_action_be_done)` could be replaced with `results doSomething(parameters_that_are_needed_for_the_action)`, with the caller deciding whether it should be called. Then, `doSomething` is reusable in another context. Ok to reject if you disagree.